### PR TITLE
remove RIODescData again, it enabled manipulating plugins private parts from outsido the plugin.

### PR DESCRIPTION
**Pull request by condret**
This is bad, since it could break the plugin itself

### CODE REVIEW

File: libr/debug/map.c [hunk 1]
```diff
@@ -300,9 +300,8 @@ R_API void r_debug_map_free(RDebugMap *map) {
 
 R_API RList *r_debug_map_list_new() {
 	RList *list = r_list_new ();
-	if (!list) {
```
Comment (radare) [inline libr/debug/map.c]: Use rlistnewf

File: libr/debug/p/debug_esil.c [hunk 2]
```diff
@@ -149,7 +149,7 @@ RDebugPlugin r_debug_plugin_esil = {
 	.wait = &__esil_wait,
 	.stop = __esil_stop,
 	.kill = __esil_kill,
-	.breakpoint = (RBreakpointCallback)&__esil_breakpoint,
+	.breakpoint = &__esil_breakpoint,
```
Comment (radare) [inline libr/debug/p/debug_esil.c]: Regression

File: libr/debug/p/debug_native.c [hunk 3]
```diff
@@ -716,8 +716,11 @@ static RList *r_debug_native_threads (RDebug *dbg, int pid) {
 #endif
 }
 
+
```
Comment (radare) [inline libr/debug/p/debug_native.c]: Empty lines

File: libr/io/p/io_mach.c [hunk 4]
```diff
@@ -197,32 +188,30 @@ static ut64 getNextValid(RIO *io, RIODesc *fd, ut64 addr) {
 	return lower;
 }
 
-static int __read(RIO *io, RIODesc *desc, ut8 *buf, int len) {
+static int __read(RIO *io, RIODesc *fd, ut8 *buf, int len) {
```
Comment (radare) [inline libr/io/p/io_mach.c]: Fd != desc aka naming regression

File: libr/io/p/io_mach.c [hunk 5]
```diff
@@ -197,32 +188,30 @@ static ut64 getNextValid(RIO *io, RIODesc *fd, ut64 addr) {
...
 	vm_size_t size = 0;
 	int blen, err, copied = 0;
 	int blocksize = 32;
-	RIODescData *dd = (RIODescData *)desc->data;
-	if (!io || !desc || !buf || !dd) {
-		return -1;
-	}
-	if (dd ->magic != r_str_hash ("mach")) {
+	RIOMach *riom;
+	if (!io || !fd || !buf || !riom) {
```
Comment (radare) [inline libr/io/p/io_mach.c]: Uninitialezed riom regression

File: libr/io/p/io_mach.c [hunk 6]
```diff
@@ -197,32 +188,30 @@ static ut64 getNextValid(RIO *io, RIODesc *fd, ut64 addr) {
...
 		return -1;
 	}
+	riom = (RIOMach *)fd->data;
```
Comment (radare) [inline libr/io/p/io_mach.c]: Riom is bad. Crashing regression

File: libr/io/p/io_mach.c [hunk 7]
```diff
@@ -197,32 +188,30 @@ static ut64 getNextValid(RIO *io, RIODesc *fd, ut64 addr) {
...
+
 	memset (buf, 0xff, len);
-	int pid = __get_pid (desc);
-	task_t task = pid_to_task (pid);
-	if (task_is_dead (pid)) {
+	if (task_is_dead (riom->pid)) {
 		return -1;
 	}
-	if (pid == 0) {
+	if (RIOMACH_PID (fd->data) == 0) {
```
Comment (radare) [inline libr/io/p/io_mach.c]: Regression

File: libr/io/p/io_mach.c [hunk 8]
```diff
@@ -197,32 +188,30 @@ static ut64 getNextValid(RIO *io, RIODesc *fd, ut64 addr) {
...
 		if (io->off < 4096) {
 			return len;
 		}
 	}
-	copied = getNextValid (io, desc, io->off) - io->off;
+	copied = getNextValid (io, fd, io->off) - io->off;
```
Comment (radare) [inline libr/io/p/io_mach.c]: Naming regression

File: libr/io/p/io_mach.c [hunk 9]
```diff
@@ -443,41 +422,29 @@ static ut64 __lseek(RIO *io, RIODesc *fd, ut64 offset, int whence) {
...
 	if (!io || !fd || !cmd || !fd->data) {
 		return 0;
 	}
-	RIODescData *iodd = fd->data;
-	if (iodd->magic != R_MACH_MAGIC) {
-		return false;
-	}
-	
-	task_t task = pid_to_task (iodd->tid);
+	riom = (RIOMach*)fd->data;
```
Comment (radare) [inline libr/io/p/io_mach.c]: Accessing private firlds regression

File: libr/io/p/io_mach.c (hunk 4 above)
Comment (condret) [reply libr/io/p/io_mach.c]: naming regression, srsly?

File: libr/io/p/io_mach.c (hunk 9 above)
Comment (condret) [reply libr/io/p/io_mach.c]: this is in the plugin, the plugin may do this

File: libr/io/p/io_mach.c (hunk 5 above)
Comment (condret) [reply libr/io/p/io_mach.c]: read the rest of the code, this line is fine

**Comment by radare** [comment]
io provides pid only.. but there must be thread id, handle for widnows and task for windows.. to be shared between io and debug, and that must be portable across different targets, thats the overall idea of riodescdata, to have a place to share opaque structs between io and debug, no one else should have this, and imho the amount of info we want to share is quite small, that it can be even serialized via io.cmd()

**Comment by radare** [comment]
can we close this pr?

**Comment by condret** [comment]
the pid should be hidden in the ioplugin and the debug should ask for it, and never set it. Why should the debugger set the pid. the pid is set by the OS and is ro, afaik

File: libr/core/cmd_debug.c [hunk 10]
```diff
@@ -4509,7 +4509,7 @@ static int cmd_debug(void *data, const char *input) {
 	if (follow > 0) {
 		ut64 pc = r_debug_reg_get (core->dbg, "PC");
 		if ((pc < core->offset) || (pc > (core->offset + follow))) {
-			r_core_cmd0 (core, "sr PC");
+			r_core_cmd0 (core, "sr PC");		//XXX
```
Comment (MaskRay) [inline libr/core/cmd_debug.c]: condret is right here. Use API, rather call r_core_cmd0.

File: libr/io/p/io_mach.c (hunk 4 above)
Comment (MaskRay) [reply libr/io/p/io_mach.c]: I prefer single underscore. Double underscores are reserved for implementations.

http://www.open-std.org/jtc1/sc22/wg14/www/docs/n1570.pdf
All identifiers that begin with an underscore and either an uppercase letter or another underscore are always reserved for any use.

**Pull request closed without merging**